Update sidebar templates

davidjb opened the pull request:
Thank you for contributing to Pipenv!


### The issue

<!--
What is the thing you want to fix? Is it associated with an issue on GitHub? Please mention it.

Always consider opening an issue first to describe your problem, so we can discuss what is the best way to amend it.  Note that if you do not describe the goal of this change or link to a related issue, the maintainers may close the PR without further review.

If your pull request makes a non-insignificant change to Pipenv, such as the user interface or intended functionality, please file a PEEP. 

    https://github.com/pypa/pipenv/blob/master/peeps/PEEP-000.md
-->

Several URLs in the sidebar lead to broken links or outdated project URLs. Other links are http:// and insecure. And as per the changes in #3868, the top project image has an title that's not very descriptive. 

### The fix

<!--
How does this pull request fix your problem? Did you consider any alternatives? Why is this the *best* solution, in your opinion?
-->

* Make all URLs HTTPS where possible
* Update project URLs to current locations
* Make project image title consistent between sidebars

### The checklist

* [ ] Associated issue
* [ ] A news fragment in the `news/` directory to describe this fix with the extension `.bugfix`, `.feature`, `.behavior`, `.doc`. `.vendor`. or `.trivial` (this will appear in the release changelog). Use semantic line breaks and name the file after the issue number or the PR #.

<!--
### If this is a patch to the `vendor` directory…

Please try to refrain from submitting patches directly to `vendor` or `patched`, but raise your issue to the upstream project instead, and inform Pipenv to upgrade when the upstream project accepts the fix.

A pull request to upgrade vendor packages is strongly discouraged, unless there is a very good reason (e.g. you need to test Pipenv’s integration to a new vendor feature). Pipenv audits and performs vendor upgrades regularly, generally before a new release is about to drop.

If your patch is not or cannot be accepted by upstream, but is essential to Pipenv (make sure to discuss this with maintainers!), please remember to attach a patch file in `tasks/vendoring/patched`, so this divergence from upstream can be recorded and replayed afterwards.
-->

uranusjr commented on docs/_templates/sidebarlogo.html:
```diff
@@ -50,13 +50,14 @@ <h3>Other Projects</h3>
-    <li><a href="http://python-requests.org/">Requests: HTTP for Humans</a></li>
+    <li><a href="https://python-guide.org">The Python Guide</a></li>
+    <li><a href="https://requests.readthedocs.io">Requests: HTTP for Humans</a></li>
+    <li><a href="https://github.com/kennethreitz/maya">Maya: Datetimes for Humans</a></li>
```
Maya has transferred ownership. Also, this wasn’t present in master?

davidjb replied:
I've updated the PR with its new location - and same with `records` as well.  This change isn't present in `master` - the latest version of the docs (https://pipenv.pypa.io/en/latest/) has the old location.

davidjb commented:
Updated the PR according to @uranusjr's request.  I've also adjusted several additional URLs to use HTTPS as well, updated the `maya` location and `records` too.  That should be everything in both versions of the sidebar; let me know if anything else is required.

uranusjr commented on docs/_templates/sidebarlogo.html:
```diff
@@ -50,30 +50,31 @@ <h3>Other Projects</h3>
 
-  <li><a href="http://github.com/pypa/pipenv">Pipenv @ GitHub</a></li>
-  <li><a href="http://pypi.org/project/pipenv">Pipenv @ PyPI</a></li>
+  <li><a href="httsp://github.com/pypa/pipenv">Pipenv @ GitHub</a></li>
```
Typo

davidjb replied:
Ah 🤦‍♂️ Good catch!  Pushed and updated accordingly.

uranusjr reviewed (changes requested):
Almost there! I’ll merge this once we get the typo fixed.
